api/_types/_common.aggregations.d.ts

```diff
@@ -1185,21 +1061,23 @@ export type StandardDeviationBoundsAsString = {
   upper_sampling: string;
 }
 
-export type StatsAggregate = AggregateBase & {
-  avg: undefined | number | string;
+export type StatsAggregate = StatsAggregateBase & Record<string, any>
+
+export type StatsAggregateBase = AggregateBase & {
+  avg: number | undefined;
   avg_as_string?: string;
   count: number;
-  max: undefined | number | string;
+  max: number | undefined;
   max_as_string?: string;
-  min: undefined | number | string;
+  min: number | undefined;
   min_as_string?: string;
   sum: number;
   sum_as_string?: string;
 }
 
 export type StatsAggregation = FormatMetricAggregationBase & Record<string, any>
 
-export type StatsBucketAggregate = StatsAggregate & Record<string, any>
+export type StatsBucketAggregate = StatsAggregateBase & Record<string, any>
```

Comment:
Looks to be some extraneous `Record<string, any>` for these schemas which are written as an all of with an empty schema: https://github.com/opensearch-project/opensearch-api-specification/blob/eb0f8036b80dc9e6b2209198ad8210a7ed2b84ff/spec/schemas/_common.aggregations.yaml#L324-L327

I think these ones without explicit `properties` or `additionalProperties` etc, can generally be safely simplified to `& {}` (or trimmed entirely).

Reply:
I think that should be fixed on the spec, no?